src/commands/CommandFactory.ts

```diff
@@ -0,0 +1,29 @@
+import { Database } from "../types/supabaseData.js";
+import { ISafeApiCommand } from "../types/safe-signatures.js";
+
+import { UserUpsertCommand } from "./UserUpsertCommand.js";
+import { SafeApiCommand } from "./SafeApiCommand.js";
+
+type SignatureRequest =
+  Database["public"]["Tables"]["signature_requests"]["Row"];
+
+export function getCommand(request: SignatureRequest): ISafeApiCommand {
+  switch (request.purpose) {
+    case "update_user_data":
+      return new UserUpsertCommand(
+        request.safe_address,
+        request.message_hash,
+        request.chain_id,
+      );
+    default:
+      return new NoopCommand();
```

Comment:
My assumption here would be that a request with an unknown purpose is an error. Is there a reason to not juss throw here?

Reply:
Hmm, that would purposefully crash the backend. I don't think that's a good idea. How about we do a console log and perhaps configure some sort of notification to discord when that log statement was created, so we know this is happening without having to crash the backend. Wdyt?

Reply:
yes to monitoring and error reporting in our Discord. On the throwing of errors, we do the same in `getRprUrl.ts` and other method. 

This is mostly a point on consistency, and I'd be happy to park that for the bigger cleanup push.

Reply:
Okay great, I'll add a `console.warning` here and once we get this into people's hands will monitor the service to make sure there are no surprises here.